### PR TITLE
Also convert first part of email to idn

### LINKED ISSUE #13586: [User management] International email addresses aren't valid
### Steps to reproduce

Using an admin user:

Activate send email to new users checkbox
1. Create a new user with international email address.
### Expected behaviour

International email addresses are allowed.
### Actual behaviour

International email addresses (with Ñ,Ö,Ç..) are invalid.
### Server configuration

**Operating system**:
Ubuntu 14.04 LTS

**Web server:**
Apache

**Database:**
MySQL

**PHP version:**
5.5.9

**ownCloud version:** (see ownCloud admin page)
{"installed":true,"maintenance":false,"version":"8.0.0.1","versionstring":"8.0 alpha 1","edition":""}

**Updated from an older ownCloud or fresh install:**
fresh

**List of activated apps:**
none

**The content of config/config.php:**

```
Insert your config.php content here
(Without the database password and passwordsalt)
```

**Are you using external storage, if yes which one:** local/smb/sftp/...
no

**Are you using encryption:** no
### Client configuration

**Browser:**
Chrome, Firefox

### PR DESCRIPTION
**Pull request by rullzer**
Fixes https://github.com/owncloud/core/issues/13586

Basically do the same trick we did for the damain.

@SergioBertolinSG since you filled https://github.com/owncloud/core/issues/13586 can you test this. My e-mail hosten apparently does not allow those kind of e-mail addresses. (so also if e-mail is properly send).

### CODE REVIEW

**Comment by DeepDiver1975** [comment]
Is idn valid for the name? afaik idn is only allowed for the domain name

**Comment by rullzer** [comment]
Yes, see for example http://tools.ietf.org/html/rfc6530 section 7.1

```Permits the use of UTF-8 strings in email addresses, both local parts and domain names.```

**Comment by DeepDiver1975** [comment]
> Yes, see for example http://tools.ietf.org/html/rfc6530 section 7.1

Nice - thx - I was search for something like this in the past but could not find it!
http://tools.ietf.org/html/rfc6530#section-7.1

**Comment by DeepDiver1975** [comment]
:+1:

File: lib/private/mail/mailer.php [hunk 1]
```diff
@@ -139,6 +139,7 @@ protected function convertEmail($email) {
 
 		list($name, $domain) = explode('@', $email, 2);
 		$domain = idn_to_ascii($domain);
+		$name = idn_to_ascii($name);
```
Comment (nickvergessen) [inline lib/private/mail/mailer.php]: As per docs of the method in php, it is for "domain name"
http://php.net/manual/en/function.idn-to-ascii.php

**Comment by nickvergessen** [comment]
I used `täst@...` and tried sending a share link:
It failed and the log entry says:
```
Can't send mail with public link to t\u00e4st@...: Address in mailbox given [t\u00e4st@...] does not comply with RFC 2822, 3.6.2
```
So this seems to be not possible with swift mailer in the version we use (5.3.1 from dec 2014).
But also the change logs don't look much promising: https://github.com/swiftmailer/swiftmailer/blob/5.x/CHANGES#L9-L22

**Comment by rullzer** [comment]
mmm so it seems we can't fix it as easily as I had hoped... seems we really do need to wait for swiftmailer 6 :(

**Comment by rullzer** [comment]
So closing this as it apparently does not work :(

**Comment by SergioBertolinSG** [comment]
BTW gmail only supports sending and receiving from those addresses but not creating one. I didn't find an email provider with that option yet.

**Pull request closed without merging**